### ISSUE TITLE
Als formulierbouwer wil ik de waarden van alle waarden zoals formuliervelden, objectregistraties, zaakgegevens, variabelen, static, et cetera, kunnen hergebruiken in alle daarna-volgende attributen en condities

### ISSUE THREAD

**Issue by joeribekker** [open-formulieren/open-forms#2168]
Als formulierbouwer wil ik de waarden van alle waarden zoals formuliervelden, objectregistraties, zaakgegevens, variabelen, static, et cetera, kunnen hergebruiken in alle daarna-volgende attributen en condities, zodat ik context kan toevoegen aan het formulier.

Oorsprong: https://github.com/Dimpact-Samenwerking/Open-Formulieren/issues/50
RTM: 4.1
Story points: 5
Business value: 5

**Comment by sergei-maertens** [comment]
**Technical analysis & implementation guidelines**

## To refine

* Upstream userstory mentions "zaakgegevens", but registration runs after submission completion, so that will never be available. We do extract case number etc. for display. Is there another need we are missing?
* "alle daarna-volgende attributen en condities" - client-side logic only ever has access to variables within the form itself, so any custom variables we add (like static variables) cannot be used there but need to be used through backend logic. Are we missing a use case here?

## Status

* We currently support templating for the following properties: `label`, `legend`, `defaultValue`, `description`, `html` and `placeholder`. Any variable that exists in the backend can be used in these templates.
* #2169 adds support for fetching data from whatever-service and storing it in a variable, thereby giving access to this data as well
* #2169 also mentions supporting dropdown/radio/multi checkbox option sources from variables
* Any defined variable can be referenced in logic rule triggers (or actions)

So in short - this should be mostly covered by #2169 and existing functionality.

**Comment by joeribekker** [comment]
> Upstream userstory mentions "zaakgegevens", but registration runs after submission completion, so that will never be available. We do extract case number etc. for display. Is there another need we are missing?

I suspect reading zaakgegevens as prefill based on a user's zaken. For example, "Does your question concern an existing Zaak? Select it here"

But this is guessing, so of course refinement is good to do.

**Comment by joeribekker** [comment]
Even bespreken of deze dicht kan.

**Comment by joeribekker** [comment]
Refinement: Dat kan hij.
